### PR TITLE
Update and fix cognee integration

### PR DESCRIPTION
**Pull request by daukadolt**
# Description

Restored and enhanced the CogneeGraphRAG integration that was previously moved to stale packages due to broken tests. This change fixes the existing test issues and adds comprehensive test coverage to ensure the integration is robust and reliable.

The restored integration now includes:
- Fixed existing test failures that caused the package to be moved to stale
- Enhanced test coverage for edge cases, error handling, and input validation
- Mocked workflows that don't require external API dependencies for CI/CD
- Improved error handling and validation

The new test suite covers:
- Empty document handling and validation
- Single document vs list processing  
- Graph visualization functionality
- Search error handling with empty results
- Complete workflow testing with mocked dependencies
- Mixed document types with edge cases

This addresses the issues mentioned in PR #19627 where tests were broken, making the integration production-ready again.

Fixes # (restores functionality from stale packages)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [X] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [X] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [X] I added new unit tests to cover this change
- [X] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I ran `uv run make format; uv run make lint` to appease the lint gods

## Restoration Details

This PR addresses the issues that caused cognee to be moved to stale packages:

1. **Fixed broken tests** - Resolved the test failures mentioned in the stale move
2. **Enhanced test coverage** - Added comprehensive tests for edge cases and error scenarios
3. **Improved reliability** - Added mocked tests that work without external dependencies
4. **Better error handling** - Ensured graceful handling of various input scenarios

The new test suite includes:

1. **`test_empty_documents()`** - Tests handling of empty document lists and validates proper error handling
2. **`test_single_document()`** - Tests processing single documents vs lists of documents
3. **`test_visualize_graph()`** - Tests graph visualization with different parameters and error conditions
4. **`test_search_error_handling()`** - Tests behavior when search methods return empty results
5. **`test_mock_full_workflow()`** - Complete workflow test with mocked dependencies for offline testing
6. **`test_mixed_document_types()`** - Tests handling of mixed valid/invalid document types

The integration is now ready for production use with reliable test coverage.

### CODE REVIEW

File: llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    "types-requests==2.28.11.8",
+    "types-setuptools==67.1.0.0",
+    "black[jupyter]<=23.9.1,>=23.7.0",
+    "codespell[toml]>=v2.2.6",
+    "diff-cover>=9.2.0",
+]
+
+[project]
+name = "llama-index-graph-rag-cognee"
+version = "0.2.1"
```
Comment (logan-markewich) [inline llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/pyproject.toml]: lets bump this to 0.3.0?

Comment (daukadolt) [reply llama-index-integrations/graph_rag/llama-index-graph-rag-cognee/pyproject.toml]: sounds good, updating

**Comment by logan-markewich** [comment]
Seems like we either need to mark some tests as skip if llm credentials are missing or mock out the llm calls?

**Comment by daukadolt** [comment]
Apologies, I forgot to skip unit tests that rely on LLM credentials, let me update the PR

**Review by logan-markewich — APPROVED** [review]

**Comment by logan-markewich** [comment]
Thanks for the efforts here!
